lib/impress.js

```diff
@@ -163,24 +164,22 @@ impress.compareHosts = () => {
   const cmp = [];
   for (appName in impress.applications) {
     config = impress.applications[appName].config;
-    if (config) {
-      hosts = config.hosts;
-      if (hosts) {
-        for (i = 0, ilen = hosts.length; i < ilen; i++) {
-          for (j = 0, jlen = cmp.length; j < jlen; j++) {
-            if (impress.compareMasks(hosts[i], cmp[j])) {
-              impress.log.warning(
-                'Hosts mask overlapping: "' + hosts[i] +
-                '" and "' + cmp[j] + '"'
-              );
-            }
-          }
-        }
-        for (k = 0, klen = hosts.length; k < klen; k++) {
-          if (!cmp.includes(hosts[k])) cmp.push(hosts[k]);
+    if (!config) continue;
+    hosts = config.hosts;
+    if (!hosts) continue;
+    for (i = 0, ilen = hosts.length; i < ilen; i++) {
+      for (j = 0, jlen = cmp.length; j < jlen; j++) {
+        if (impress.compareMasks(hosts[i], cmp[j])) {
+          impress.log.warning(
+            'Hosts mask overlapping: "' + hosts[i] +
+            '" and "' + cmp[j] + '"'
+          );
         }
       }
     }
+    for (k = 0, klen = hosts.length; k < klen; k++) {
```

Comment:
As a side-effect, it will also report if there are hosts overlap inside a config of a single application. If that's not desirable, `hosts[i]`s can be pushed into a new array and then `.concat`ed with `cmp`.